homeworkW12D03/src/ba/bitcamp/homeworkW12D03/task02/PicChangerGUI.java

```diff
@@ -0,0 +1,178 @@
+
+	private static final long serialVersionUID = -1592269176478889092L;
+
+	// Declaring main elements.
+	private JPanel panel = new MyPanel();
+	private JButton button = new JButton("Negative image");
+
+	// Declaring list of tasks and list of consumers.
+	private LinkedBlockingQueue<Producer> tasks = new LinkedBlockingQueue<>();
+	private ArrayList<Consumer> consumers = new ArrayList<>();
+
+	// Declaring variable that will read image.
+	private BufferedImage image;
+
+	public PicChangerGUI() {
+
+		// Setting the look of panel according to system look.
+		try {
+			UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+		} catch (ClassNotFoundException | InstantiationException | IllegalAccessException | UnsupportedLookAndFeelException e2) {
+			e2.printStackTrace();
+		}
+		// Reading image.
+		try {
+			image = ImageIO.read(new File("picture.jpg"));
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		// Adding action listener to the button and setting the font of the
+		// button text.
+		button.addActionListener(this);
+		button.setFont(new Font("Serif", Font.BOLD, 20));
+
+		// Setting the layout of the main panel and adding elements.
+		setLayout(new BorderLayout());
+		add(panel);
+		add(button, BorderLayout.SOUTH);
+	}
+
+	/**
+	 * Divides the tasks into parts and starts consumers.
+	 */
+	public void actionPerformed(ActionEvent e) {
+		// Allocating one row of pixels to one producer and adding producer to
+		// the list.
+		for (int i = 0; i < image.getHeight(); i++) {
+			Producer p = new Producer(image.getWidth(), i);
+			tasks.add(p);
+		}
+		// Creating, starting and adding consumers to the list.
+		for (int i = 0; i < 4; i++) {
+			Consumer con = new Consumer();
+			con.start();
+			consumers.add(con);
+		}
+		// Changing the text of the button depending on the state of the image.
+		if (button.getText().equals("Negative image")) {
+			button.setText("Positive image");
+		} else {
+			button.setText("Negative image");
+		}
+	}
+
+	/**
+	 * Returns the imported image.
+	 * 
+	 * @return Image that is imported.
+	 */
+	public BufferedImage getImage() {
+		return image;
+	}
+
+	/**
+	 * 
+	 * @author Gordan
+	 *
+	 *         Inner class represents JPanel that draws imported image.
+	 */
+	private class MyPanel extends JPanel {
+
+		private static final long serialVersionUID = 3873896819533186187L;
+
+		protected void paintComponent(Graphics g) {
+			super.paintComponent(g);
+			g.drawImage(image, 0, 0, null);
+			repaint();
+		}
+	}
+
+	/**
+	 * 
+	 * @author Gordan
+	 * 
+	 *         Inner class that represents task that is need to be executed.
+	 *
+	 */
+	private class Producer implements Runnable {
+
+		// Properties of the producer.
+		private int row;
+		private int column;
+
+		/**
+		 * Constructor that creates new producer.
+		 * 
+		 * @param row
+		 *            - Number of rows.
+		 * @param column
+		 *            - Number of columns.
+		 */
+		public Producer(int row, int column) {
```

Comment:
this should be something like `public Producer(int row, int width)` or even omit the width since you do have access to the width attribute from the `BufferedImage` object.